src/search-headless.ts

```diff
@@ -555,5 +558,74 @@ export default class SearchHeadless {
   setFilterOption(filter: SelectableStaticFilter): void {
     this.stateManager.dispatchEvent('filters/setFilterOption', filter);
   }
+
+  /**
+   * Perform a generativeDirectAnswer request to the query most recent search stored in state.
+   *
+   * @returns A Promise of a {@link GenerativeDirectAnswerResponse} from the Search API or
+   *          of undefined if there is no results defined in state
+   */
+  async executeGenerativeDirectAnswer(): Promise<GenerativeDirectAnswerResponse | undefined> {
+    const searchId = this.state.meta.uuid;
+    if (!searchId) {
+      console.error('no search id supplied for generative direct answer');
+      return;
+    }
+    let requestName: 'universalQuery' | 'verticalQuery';
+    let results: VerticalResults[] | undefined;
+    if (this.state.meta.searchType === SearchTypeEnum.Vertical) {
+      requestName = 'verticalQuery';
+      if (this.state.vertical.appliedQueryFilters && this.state.vertical.queryDurationMillis 
+        && this.state.vertical.results && this.state.vertical.resultsCount 
+        && this.state.vertical.source && this.state.vertical.verticalKey ) {
+        results = [{
+          appliedQueryFilters: this.state.vertical.appliedQueryFilters,
+          queryDurationMillis: this.state.vertical.queryDurationMillis,
+          results: this.state.vertical.results,
+          resultsCount: this.state.vertical.resultsCount,
+          source: this.state.vertical.source,
+          verticalKey: this.state.vertical.verticalKey,
+        }];
+      }
+    } else {
+      requestName = 'universalQuery';
+      results = this.state.universal.verticals;
+    }
+    if (!results || results.length === 0) {
+      console.error('no results supplied for generative direct answer');
+      return;
+    }
+    const searchTerm = this.state.query.mostRecentSearch;
+    if (!searchTerm) {
+      console.error('no search term supplied for generative direct answer');
+      return;
+    }
+    const thisRequestId = this.httpManager.updateRequestId(requestName);
+    this.stateManager.dispatchEvent('generativeDirectAnswer/setIsLoading', true);
+
+    let response: GenerativeDirectAnswerResponse;
+    try {
+      response = await this.core.generativeDirectAnswer({
+        searchId,
+        results,
+        searchTerm,
+        additionalHttpHeaders: this.additionalHttpHeaders
+      });
+    } catch (e) {
+      const isLatestResponse = this.httpManager.processRequestId(requestName, thisRequestId);
```

Comment:
the intent of httpManager/processRequestId is to ensure the response is the latest response correspond to the request type (see [file](https://github.com/yext/search-headless/blob/main/src/http-manager.ts)). So the `requestName` shouldn't be universalQuery or verticalQuery, since that would affect those request types. You should provide a unique request to track GDA requests specifically